src/storage/heap_file.c

```diff
@@ -20091,7 +20201,8 @@ heap_update_adjust_recdes_header (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEX
 	}
     }
 
-  if (is_mvcc_op)
+#if defined (SERVER_MODE)
+  if (is_mvcc_class && !is_update_in_place)
```

Comment:
Good idea. 
It would be great to also replace if clause of heap_update_logical with the macro.